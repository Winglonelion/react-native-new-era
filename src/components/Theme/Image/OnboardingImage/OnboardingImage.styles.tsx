import { StyleSheet } from 'react-native';
import { responsiveHeight } from 'utils/screen';

const styles = StyleSheet.create({
  image: {
    width: responsiveHeight(280),
    resizeMode: 'contain',
    height: responsiveHeight(280),
  },
});

export default styles;
