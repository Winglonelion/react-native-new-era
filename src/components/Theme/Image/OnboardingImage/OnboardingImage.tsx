import React from 'react';
import { Image, ImageProps } from 'react-native';
import styles from './OnboardingImage.styles';

interface PropTypes extends ImageProps {}

const OnboardingImage: React.FC<PropTypes> = ({ source, style }) => {
  return <Image source={source} style={[styles.image, style]} />;
};

export default OnboardingImage;
