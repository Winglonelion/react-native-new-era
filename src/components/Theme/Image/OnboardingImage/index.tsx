export { default as default } from './OnboardingImage';
