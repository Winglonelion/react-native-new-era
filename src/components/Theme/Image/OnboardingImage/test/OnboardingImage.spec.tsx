import React from 'react';
import { shallow } from 'enzyme';
import OnboardingImage, { PropTypes } from '../OnboardingImage';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<OnboardingImage {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('OnboardingImage check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      source: require('images/account_info.png'),
      style: { width: 30, height: 30 },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
