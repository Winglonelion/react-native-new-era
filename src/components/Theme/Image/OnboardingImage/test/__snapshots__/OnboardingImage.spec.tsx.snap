// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingImage check snapshot should match snapshot with props 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/images/account_info.png",
    }
  }
  style={
    Array [
      Object {
        "height": 460,
        "resizeMode": "contain",
        "width": 460,
      },
      Object {
        "height": 30,
        "width": 30,
      },
    ]
  }
/>
`;
